packages/react/src/primitives/Divider/Divider.tsx

```diff
@@ -6,14 +6,19 @@ import { DividerProps, Primitive } from '../types';
 import { View } from '../View';
 
 const DividerPrimitive: Primitive<DividerProps, 'hr'> = (
-  { className, orientation = 'horizontal', size, ...rest },
+  { className, orientation = 'horizontal', size, label, ...rest },
   ref
 ) => (
   <View
     aria-orientation={orientation}
     as="hr"
-    className={classNames(ComponentClassNames.Divider, className)}
+    className={classNames(
+      ComponentClassNames.Divider,
+      className,
+      label ? ComponentClassNames.DividerLabel : null
+    )}
     data-size={size}
+    data-label={label}
```

Comment:
Note: we do need to keep this data attribute around because we use it in CSS with the `attr()` function.